qiskit/circuit/quantumcircuit.py

```diff
@@ -1406,6 +1529,273 @@ def _update_parameter_table(self, instruction: CircuitInstruction):
                     # clear cache if new parameter is added
                     self._parameters = None
 
+    @typing.overload
+    def get_var(self, name: str, default: T) -> Union[expr.Var, T]:
+        ...
+
+    # The builtin `types` module has `EllipsisType`, but only from 3.10+!
+    @typing.overload
+    def get_var(self, name: str, default: type(...) = ...) -> expr.Var:
+        ...
+
+    # We use a _literal_ `Ellipsis` as the marker value to leave `None` available as a default.
```

Comment:
Fwiw I did this rather than the more standard pattern of an `object()` as the default value so that the Sphinx docs wouldn't look gross.